ntoskrnl/io/pnpmgr/pnpreport.c

```diff
@@ -527,16 +528,20 @@ IoReportTargetDeviceChangeAsynchronous(IN PDEVICE_OBJECT PhysicalDeviceObject,
         return STATUS_INVALID_DEVICE_REQUEST;
     }
 
+    /* Calculate the required size for the work item and notification structure */
+    WorkItemSize = FIELD_OFFSET(INTERNAL_WORK_QUEUE_ITEM, NotificationStructure)
+                   + notifyStruct->Size;
```

Comment:
Directly using the given size, or should we do here: `max(notifyStruct->Size, sizeof(TARGET_DEVICE_CUSTOM_NOTIFICATION))`, in case the caller sets the Size to zero?

Reply:
I'd rather check/assert that the size is valid.